pexec/process.go

```diff
@@ -13,13 +13,15 @@ import (
 
 // A ProcessConfig describes how to manage a system process.
 type ProcessConfig struct {
-	ID        string    `json:"id"`
-	Name      string    `json:"name"`
-	Args      []string  `json:"args"`
-	CWD       string    `json:"cwd"`
-	OneShot   bool      `json:"one_shot"`
-	Log       bool      `json:"log"`
-	LogWriter io.Writer `json:"-"`
+	ID          string    `json:"id"`
+	Name        string    `json:"name"`
+	Args        []string  `json:"args"`
+	CWD         string    `json:"cwd"`
+	OneShot     bool      `json:"one_shot"`
+	Log         bool      `json:"log"`
+	LogWriter   io.Writer `json:"-"`
+	StopSignal  string    `json:"stop_signal"`
+	StopTimeout float64   `json:"stop_timeout_seconds"`
```

Comment:
better to convert to a duration

Reply:
It's converted to a duration later, but perhaps I'm missing something to let me do it earlier. But from my understanding...

There's no native unmarshaller for time.Duration. It results in the underlying int64 type, so can't use decimal figures (I wanted seconds to be a fraction), and we definitely don't want users to have to input nanoseconds into a config.

Is there something I'm missing that makes this easier?

Reply:
Yes you can make your own unmarshal function. See https://github.com/viamrobotics/rdk/blob/main/config/config.go#L490.